runtime/common/KernelWrapper.h

```diff
@@ -0,0 +1,347 @@
+/****************************************************************-*- C++ -*-****
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+#pragma once
+#include "cudaq/platform.h"
+#include "cudaq/utils/registry.h"
+#include <cstdint>
+#include <cstring>
+#include <string>
+#include <vector>
+
+/*! \file KernelWrapper.h
+    \brief Utility classes to support library-mode kernel launch.
+
+    This header file defines classes supporting launching quantum kernels in
+   library mode (without the bridge) when using the remote platform.
+    Specifically, it provides a layer of indirection, `cudaq::invokeKernel`,
+   similar to the `altLaunchKernel` function when using the bridge. In addition,
+   argument packing (for serialization) is handled by C++ templates in lieu of
+   bridge-generated functions. These utilities will only be used for the remote
+   platform target in library-mode (guarded by pre-defined macros).
+*/
+
+namespace cudaq {
+
+/// @brief Wrapper for kernel address
+///
+/// This automatically handles kernels defined as struct/class or free
+/// functions.
```

Comment:
Do you have to consider the distinction between a free or static function vs. an OO function with a `this` pointer?

Reply:
This util class is actually no longer needed. I've removed it accordingly.